esiclient/v1/node_network.py

```diff
@@ -54,23 +54,46 @@ def take_action(self, parsed_args):
         if parsed_args.network:
             filter_network = neutron_client.find_network(parsed_args.network)
 
+        nodes = ironic_client.node.list()
+        neutron_ports = neutron_client.ports()
+        networks = neutron_client.networks()
+
         data = []
+        node_name = ''
         for port in ports:
-            node = ironic_client.node.get(port.node_uuid)
+            if not parsed_args.node or not node_name:
+                for node in nodes:
+                    if node.uuid == port.node_uuid:
+                        node_name = node.name
+                        break
+
```

Comment:
I think you can reduce the lines by searching the list of dicts as described here: https://stackoverflow.com/questions/8653516/python-list-of-dictionaries-search